metaci/api/tests/test_api_robot.py

```diff
@@ -0,0 +1,149 @@
+"""Test cases for /api/robot, mostly focusing on csv output"""
+
+from rest_framework.test import APIClient, APITestCase
+from metaci.conftest import (
+    BranchFactory,
+    RepositoryFactory,
+    UserFactory,
+)
+from metaci.testresults.models import TestResult
+from metaci.conftest import StaffSuperuserFactory, TestResultFactory
+
+
+class TestAPIRobot(APITestCase):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.superuser = StaffSuperuserFactory()
+        cls.user = UserFactory()
+        cls.client = APIClient()
+
+        # One apex test, just to make sure it doesn't appear in any test results
+        TestResultFactory(
+            method__testclass__test_type="Apex", outcome="Pass",
+        )
+
+        repo1 = RepositoryFactory(name="repo1")
+        repo2 = RepositoryFactory(name="repo2")
+        master = BranchFactory(name="master")
+        feature = BranchFactory(name="feature/robot")
+        for (repo, source, outcome, branch, test_name, robot_keyword, message) in (
+            (repo1, "file1.robot", "Pass", master, "Passing 1", None, None),
+            (repo1, "file1.robot", "Pass", master, "Passing 2", None, None),
+            (repo2, "file2.robot", "Fail", feature, "Failing 1", "KW1", "epic fail",),
+            (repo2, "file2.robot", "Fail", feature, "Failing 2", "KW1", "epic fail",),
+            (repo2, "file3.robot", "Fail", feature, "Failing 3", "KW2", "epic fail",),
+        ):
+            TestResultFactory(
+                method__testclass__test_type="Robot",
+                build_flow__build__repo=repo,
+                build_flow__build__branch=branch,
+                method__name=test_name,
+                outcome=outcome,
+                source_file=source,
+                robot_keyword=None,
+            )
+
+    def test_superuser_access(self):
+        """Make sure the superuser can access the API"""
+        self.client.force_authenticate(self.superuser)
+        response = self.client.get("/api/robot/")
+        assert response.status_code == 200
+
+    def xtest_user_access(self):
+        """
+        Make sure the superuser can access the API
+
+        This test fails. Am I doing the test wrong or did
+        I implement the API wrong?
```

Comment:
The status code is 403 - Forbidden

Reply:
I was confused when I looked at this before because the docstring says "make sure the superuser can access the API" but it's actually testing that a NON-superuser can.

DRF views that don't explicitly specify `permission_classes` will use the default permission classes from the settings: https://github.com/SFDO-Tooling/MetaCI/blob/master/config/settings/base.py#L394 -- if this view should be more widely available, it needs to override permission_classes.